### ISSUE TITLE
Add way to manage BtrFS quotas automatically from LXC config

### ISSUE THREAD

**Issue by Harvie** [lxc/lxc#1002]
Hello,
i use BtrFS quotas to limit disk usage inside LXC. Basically as described here: https://btrfs.wiki.kernel.org/index.php/Quota_support
(i have common filesystem for all containers and i limit space for particular chroots using btrfs quotas)

However i have to manage quotas manually. It would be super cool to just set something like `lxc.rootfs.btrfs.quota = 20G` in LXC config and have lxc to automatically check if there's btrfs quota and create it if it's not there and/or change it if it's set to different value.

This should be pretty easy to implement in userspace lxc utils (just few calls to btrfs binary before starting containers). it also works with container already started as btrfs quotas can be set even when in use.

**Comment by stgraber** [comment]
It's not so easy if the rootfs is shared with multiple containers, or isn't on its own subvolume, or if btrfs doesn't have quotas enabled or if you have a quota hierarchy setup.

I'm tempted to say that this is out of scope for LXC itself as there is no precedence for us doing any such thing for any of our supported backends. It is however in scope for higher level management tools like LXD which in fact supports precisely this feature when used with btrfs.

**Comment by Harvie** [comment]
Well, i think it would make sense to do this at lowest possible level. I am not sure if LXD works on top of LXC, but i think it's the case. LXC already limits amount of ram, time and number of CPUs and pretty much any hardware resource available to container. So i think it would really make sense to even have some basic way to limit diskspace in the same place.

**Comment by stgraber** [comment]
In fact, LXC doesn't know what memory limits are. The lxc.cgroup keys are basically straight filesystem access, there is no parsing or validation of those at all in LXC.

**Comment by Harvie** [comment]
And what's the difference between pasing value to cgroup and pasing quota value to path of btrfs? There may be few differences, but the code should be still pretty much straightforward...

**Comment by Evidlo** [comment]
There is some precedence with the LVM backend and the `--fssize` option during creation.  Maybe `--fssize` could also be used when the btrfs backend is used?

**Comment by brauner** [comment]
Quota support as of now is not in the scope of LXC. We'd likely review a patch but I suggest you look into using @lxc/lxd which has quota support built in.
